### ISSUE TITLE
Update free-forecast mode with UFS_UTILS/chgres_cube integration and resolve know mode bugs

### ISSUE THREAD

**Issue by KateFriedman-NOAA** [NOAA-EMC/global-workflow#178]
Updates in this issue will do the following:
- resolve existing free-forecast mode bugs
- integrate free-forecast mode with chgres_cube gdas_init utility in UFS_UTILS
- produce free-forecast ICs from v16 retrospective parallels or operational output for GFS versions 12, 13, 14, 15, and 16

The following changes are incoming with this work:

- New:
  - add MODE variable (either "free" or "cycled") to config.base and setup scripts
  - add RETRO variable in config.getic (set YES to use v16 pre-implementation parallel output as source for init, NO=ops)
  - add SDATE check in config.base: if earlier than 2019020100 then turn off waves (no rtofs support in GDA before then)

- Rework/adjustments:
  - replace fv3ic job with init job (new init.sh and config.init) - job will run chgres_cube via UFS_UTILS gdas_init utility
  - updated UFS_UTILS version
    - checkout version in checkout.sh (new tag: ufs_utils_1_4_0)
    - update build_ufs_utils.sh to use newer UFS_UTILS build script (build_all_ufs_utils.sh -> build_all.sh)
    - update link_fv3gfs.sh to adjust ufs_utils.fd fix path and remove other exec/script symlinks that are no longer needed
  - rewrite getic job to pull warm starts or chgres_cube inputs for init job off HPSS; includes pull of pgrb anl files for verification (v14+)
  - expand IAU on/off check in config.base to include MODE check
  - low resolution resource adjustments; provided by Fanglin Yang and Jun Wang while testing getic/init jobs for all resolutions
  - remove options in checkout/build for pre-CCPP GFSv16 FV3 (CCPP is now default)
    - checkout:
      - remove "-c" argument from checkout.sh (which set run_ccpp=YES)
      - remove checkout of v16 FV3 ops tag (when run_ccpp=NO)
    - build:
      - remove "-c" argument from build_all.sh, remove check of RUN_CCPP in build_all.sh for build_fv3.sh
      - remove check of RUN_CCPP in build_fv3.sh and option to build v16 ops tag (CCPP build without waves is default now)
  - add hpc/1.1.0 and cmake module loads to machine-setup.sh for Hera and WCOSS-Dell
  - disconnect HPSS access to whether the arch jobs are added the workflow, always have archive jobs now regardless 

- Bug fixes:
  - add check for prior day rtofs symlink in waveprep job (not a bug in cycled mode)
  - adjust gfs archive tarballs depending on MODE; resolves free-forecast archive bug
  - update config.nsst to set NST_SPINUP=1 when CDATE<2017072000; bug fix provided by Xu Li
  - add check of DOBLL_WAV=YES in wavepostpnt job to resolve bug when DOBLL_WAV=NO and a mismatch is produced with the CFP options
  - add missing jobs to free-forecast mode (bufrsnd, awips*, gempak*, wavepost*)

Updated job sequence for free-forecast mode:

1. gfsgetic  - pull warm starts or chgres_cube inputs*
2. gfsinit - will use files pulled by getic job to run UFS_UTILS gdas_init utility to produce user-defined resolution forecast ICs
-- rest of jobs unchanged from current system - except for bug fixes in arch --
3. gfsfcst
4. gfspost*
5. gfsvrfy
6. gfsmetp*
6. additional downstream jobs; switch dependent
7. gfsarch

*not added to workflow on machines without HPSS access (e.g. Orion); users will need to provide inputs for init job

Thank you to @GeorgeGayno-NOAA for the UFS_UTILS updates to support this!

**Comment by KateFriedman-NOAA** [comment]
Created branch in my fork for this work: [issue178](https://github.com/KateFriedman-NOAA/global-workflow/tree/issue178)

Cloning branch and setting up test on Hera.

**Comment by KateFriedman-NOAA** [comment]
Expanding issue to cover needed updates for UFS_UTILS chgres_cube and gdas_init utility. Also adjustments are needed for some UFS_UTILS codes moving to hpc-stack (e.g. nemsio tools).

Investigate going to UFS_UTILS hpc-stack branch instead of just develop branch.

**Comment by KateFriedman-NOAA** [comment]
UFS_UTILS issue related to this one: https://github.com/NOAA-EMC/UFS_UTILS/issues/102

George Gayno's fork branch for the UFS_UTILS side: https://github.com/GeorgeGayno-NOAA/UFS_UTILS/tree/feature/gdas_init.v16

**Comment by KateFriedman-NOAA** [comment]
Wrapping up work but waiting on UFS_UTILS commit after March 17th GFSv16 implementation.
